src/python/evaluation/xlsx_run_tool.py

```diff
@@ -0,0 +1,151 @@
+                             'will be stored.'
+                             'Default is \"hyperstyle/src/python/evaluation/results\"',
+                        default=None,
+                        type=str)
+
+    parser.add_argument('--file_name', '--file_name',
+                        help='Filename for that will be created to store inspection results.'
+                             'Default is \"results.xlsx\"',
+                        default='results.xlsx',
+                        type=str)
+
+    parser.add_argument('-f', '--format',
+                        default=OutputFormat.JSON.value,
+                        choices=OutputFormat.values(),
+                        type=str,
+                        help='The output format of inspectors traceback. '
+                             'Default is JSON.'
+                             'More details on output format options in README.md')
+
+
+def create_dataframe(config) -> pd.DataFrame:
+    report = pd.DataFrame(
+        {
+            "language": [],
+            "code": [],
+            "grade": [],
+        },
+    )
+
+    dataframe = pd.read_excel(config.get_data_path())
+
+    temp_dir_path = MAIN_FOLDER.parent / 'evaluation/temporary_files'
+    lang_suffixes = {'python3': '.py', 'java8': '.java', 'java11': '.java', 'kotlin': '.kt'}
+
+    for lang, code in zip(dataframe['lang'], dataframe['code']):
+        temp_file_path = os.path.join(temp_dir_path, ('file' + lang_suffixes[lang]))
+
+        with open(temp_file_path, 'w+') as file:
+            file.writelines(code)
+            try:
+                assert os.path.exists(temp_file_path)
+
+            except AssertionError:
+                logger.exception('Path does not exist.')
+                return 2
+
+            command = config.build_command(temp_file_path, lang)
+
+        results = run_in_subprocess(command)
+        os.remove(temp_file_path)
+
+        # this regular expression matches final tool grade: EXCELLENT, GOOD, MODERATE or BAD
+        regex_match = re.match(r'^.*{"code":\s"([A-Z]+)"', results).group(1)
+
+        output = regex_match
+        if config.get_traceback():
+            output = results
+
+        report = report.append(pd.DataFrame(
+            {
+                "language": [lang],
+                "code": [code],
+                "grade": [output],
+            },
+        ))
+
+    return report
+
+
+def main() -> int:
+    parser = argparse.ArgumentParser()
+    configure_arguments(parser)
+
+    try:
+        args = parser.parse_args()
+        config = ApplicationConfig(args)
+
+        workbook = Workbook()
+        workbook_path = config.get_file_path()
+        workbook.save(workbook_path)
+
+        results = create_dataframe(config)
+        print(results, workbook_path)
+
+        with pd.ExcelWriter(workbook_path, engine='openpyxl', mode='a') as writer:
+            results.to_excel(writer, sheet_name='inspection_results', index=False)
```

Comment:
I think you should create a set of functions in xlsx_util.py for working with xlsx files